docs/guides/debugging.md

```diff
@@ -0,0 +1,91 @@
+# Debugging
+
+## Debugging? What it is and why?
+
+Debugging is one of **the most important things on C**, not only Concord. It is important **because it shows where the error** (AKA segmentation fault or core dump) **came from** since you cannot know it with ease without it.
+
+We will be focusing on two popular tools for debugging, `GDB` and `Valgrind`.
+
+You could say "`But doesn't the compiler already tell the errors?`". Well, it does, but it is not perfect, and things like WebSocket messages cannot predict and say if it's going to work fine, or it's going to fail, don't trust blindly it.
```

Comment:
Now, you could say "`But, doesn't the compiler already tell me the errors?`". Well, it does, but it is not perfect; things like WebSocket messages are completely uncheckable by the compiler.